lib/hipe/llvm/hipe_llvm_main.erl

```diff
@@ -301,12 +293,14 @@ get_liveroots(<<Root:?bits(?LR_STKINDEX_SIZE)/integer-little,
                 MoreRoots/binary>>, Acc) ->
   get_liveroots(MoreRoots, [Root | Acc]).
 
-%% @doc Extracts a bunch of integers (safepoint offsets) from a binary. Returns
-%%      a tuple as need for stack descriptors.
-get_spoffs(<<>>, Acc) ->
-  lists:reverse(Acc);
-get_spoffs(<<SPOff:?bits(?SP_ADDR_SIZE)/integer-little, More/binary>>, Acc) ->
-  get_spoffs(More, [SPOff | Acc]).
+-ifdef(BIT32).
+%% ELF32 x86 uses implicit addends.
+get_reloc_addends(Table, _Relocs) ->
+  [Add || <<Add:?bits(?SP_ADDR_SIZE)/little>> <= Table].
+-else.
+%% ELF64 x64 uses explicit addends.
+get_reloc_addends(_Table, Relocs) -> elf_format:get_rela_addends(Relocs).
```

Comment:
I would write this in two lines rather than one for uniformity with the 32-bit case.